### PR TITLE
feat: run act testrunner against axe-core

### PR DESCRIPTION
**Pull request by jeeyyy**
- This PR runs ACT testrunner against axe-core rules.
- This PR of ACT testrunner https://github.com/auto-wcag/testrunner/pull/2 needs to be merged before this one, to allow to get the latest code.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    "type": "git",
+    "url": "git+https://github.com/dequelabs/axe-act-testrunner.git"
+  },
+  "keywords": [
+    "axe-core",
+    "auto-wcag",
+    "act",
+    "testrunner"
+  ],
+  "author": "",
```
Comment (WilcoFiers) [inline package.json:21]: Feel free to put your name here Jey.

File: src/evaluate.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+/* global testcase, getTestCaseStatus */
+const axe = require('axe-core')
+const { rulesMap } = require('./rules-map')
+
+// Function to be evaluated with in the page context
+function evaluate() {
+  return new Promise((resolve, reject) => {
```
Comment (WilcoFiers) [inline src/evaluate.js:7]: axe-core returns a promise, so you can chain on that:

```js
return axe.run(document, context)
  .then(getActResult)
```

File: src/utils.js [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+}
+
+/**
+ * Get testcase status
+ * @param {Object} param composite object with attributes to compute testcase status
+ * @property {Object} param.result result of testcase
+ * @property {Object} param.testcase testcase
+ * @returns {String} status of testcase
+ */
+function getTestCaseStatus({ result, testcase }) {
```
Comment (WilcoFiers) [inline src/utils.js:71]: This function should be part of the testrunner. The test engine should return its findings to the testrunner, the test runner should decide the status. The rules for deciding the status shouldn't change between implementors.

File: src/reporter.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+ * @returns {Promise} an asynchronous function that generates reports
+ */
+async function generateReports({ types, fileName, outputDirectory, results }) {
+  if (!types || !types.length) {
+    throw new Error(
+      'Log: TestRunner: No report types are provided. Allowed types are "json" or "markdown"'
+    )
+  }
+
+  const fileData = await getFormattedResults(results)
```
Comment (WilcoFiers) [inline src/reporter.js:22]: We'll need the testrunner to generate some sort of standardized JSON. It'll have to include more than just the results. I suggest having the testrunner generate it, and from there we can generate the MD file here. It should look something like this:

```js
{
  testSystem: {
    name: String,
    version: String,
    url: String
  },
  creator: {
    name: String,
    url: String
  },
  testEnvironment: {
    name: String,
    version: String,
    ...
  },
  ruleStatus: [
    ruleId: String,
    status: Enum,
    testCases: {
      [testcaseId]: Enum #status
    }
  ]
}
```

File: package.json [hunk 5]
```diff
@@ -0,0 +1,45 @@
+{
+  "name": "axe-act-testrunner",
+  "version": "1.0.0",
+  "description": "axe-core testrunner using the ACT test case format",
+  "main": "src/index.js",
+  "scripts": {
```
Comment (WilcoFiers) [inline package.json:6]: Be sure to add some tests to this repo at some point.

**Review by WilcoFiers — APPROVED** [review]
Mostly just suggestions for future updates.

**Comment by jeeyyy** [comment]
@WilcoFiers  - I will merge this, but work on the above comments on a separate PR.

**Pull request merged**